### PR TITLE
feat: remove verbose output from Docker container logs

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

* It removes the curl output from the Docker container logs
* It removes API Key and enroll response from Docker container logs, it is possible to set the environment variable `DEBUG=1`to see these messages.

## Why is it important?

The curl downloading output is not shown correctly on logs, it does not apport value. The API key and the enroll response is sensible information so we should remove it from the logs.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- ~[ ] I have made corresponding changes to the documentation~
- ~[ ] I have made corresponding change to the default configuration files~
- ~[ ] I have added tests that prove my fix is effective or that my feature works~
- ~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~

### CODE REVIEW

File: dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl [hunk 1]
```diff
@@ -12,12 +12,23 @@ set -eo pipefail
 # KIBANA_PASSWORD - password for accessing kibana API [changeme]
 # KIBANA_USERNAME - username for accessing kibana API [elastic]
 
+function log(){
+  local msg=$1
+  if [ "${DEBUG}" == "1" ]; then
```
Comment (mdelapenya) [inline dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl:17]: Can we document this variable above too?

**Review by mdelapenya — APPROVED** [review]

File: dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl [hunk 2]
```diff
@@ -12,12 +12,23 @@ set -eo pipefail
 # KIBANA_PASSWORD - password for accessing kibana API [changeme]
 # KIBANA_USERNAME - username for accessing kibana API [elastic]
 
+function log(){
```
Comment (v1v) [inline dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl:15]:
```suggestion
function debug(){
```

File: dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl [hunk 3]
```diff
@@ -27,23 +38,27 @@ function enroll(){
...
         -H 'kbn-xsrf: true' \
         -u ${KIBANA_USERNAME:-elastic}:${KIBANA_PASSWORD:-changeme} )
+      echo ""
 
       local exitCode=$?
       if [ $exitCode -ne 0 ]; then
         exit $exitCode
       fi
-      echo $enrollResp
+      log $enrollResp
```
Comment (v1v) [inline dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl:52]:
```suggestion
      debug $enrollResp
```

File: dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl [hunk 4]
```diff
@@ -27,23 +38,27 @@ function enroll(){
...
       local apikeyId=$(echo $enrollResp | jq -r '.list[0].id')
-      echo $apikeyId
+      log $apikeyId
```
Comment (v1v) [inline dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl:54]:
```suggestion
      debug $apikeyId
```

File: dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl [hunk 5]
```diff
@@ -52,7 +67,7 @@ function enroll(){
 
       apikey=$(echo $enrollResp | jq -r '.item.api_key')
     fi
-    echo $apikey
+    log $apikey
```
Comment (v1v) [inline dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl:70]:
```suggestion
    debug $apikey
```

**Review by v1v — APPROVED** [review]
NIT, regarding the `log` method to be renamed to `debug`

**Review by cachedout — APPROVED** [review]

**Pull request closed without merging**